tests/test_other.py

```diff
@@ -7728,7 +7754,8 @@ def test_binaryen_opts(self):
           print(args, 'js opts:', expect_js_opts, 'only-wasm:', expect_only_wasm, '   ', ' '.join(cmd))
           err = run_process(cmd, stdout=PIPE, stderr=PIPE).stderr
           assert expect_js_opts == ('applying js optimization passes:' in err), err
-          assert expect_only_wasm == ('-emscripten-only-wasm' in err and '--wasm-only' in err), err # check both flag to fastcomp and to asm2wasm
+          if not self.is_wasm_backend():
```

Comment:
the whole test is no-wasm-backend, though, why is this needed? preparation for eventually enabling parts of it?

Reply:
Way I read it, the test is currently relying on asm2wasm-specific behavior. But this part should never happen in wasm backend, so it makes sense to make that expectation explicit.

Reply:
I see, sounds good.

Reply:
Yes, I initially tried to make this test pass with wasm backend, but gave up at some point. But I think this could still be useful going forward. I could revert this part I suppose.

Reply:
No need to revert, lgtm as is.